### DISCUSSION TITLE
feature request: make Ctrl-Z undo Delete Page

### DISCUSSION THREAD

**Discussion by nbeaver** [zim-desktop-wiki/zim-desktop-wiki#1707] [Ideas]
# Steps to reproduce

1. Make a test page.

2. Right-click on page name in Index → "Delete Page"

3. "Edit" → "Undo"

# Expected result

Last deleted page is restored.

# Actual result

Nothing happens. (Last edit on focused page may be reverted.)

# Discussion

In the right-click menu for pages, "Delete Page" is right below "Rename or move Page...", and sometimes I will accidentally move down too far and hit "Delete Page." Fortunately, Zim uses Gio's [trash() function](https://docs.gtk.org/gio/vfunc.File.trash.html) instead of just deleting it outright, so this isn't generally a data loss issue. (I also keep my Zim wiki folder under version control so this provides another layer of protection, as do some syncing clients.)

However, there are some drawbacks:

1. Deleting a page also deletes all the links to that page. Restoring the file does not restore links to that page.

2. Trash capability is not always available, e.g. on a networked drive.

3. Sometimes the trash daemon fails silently. On Linux, for example, gvfsd-trash can crash [or become unresponsive](https://gitlab.gnome.org/GNOME/gvfs/-/issues/485).

4. Needing to navigate to the trash folder or use version control is inconvenient and a bit opaque to new users.

Therefore, it would be nice if Zim could provide the capability to undo the last page deletion. I think the simplest way from a user's perspective would be to use Ctrl-Z / Edit → Undo.

I understand that this capability is not simple from an implementation perspective, but I think getting it on the wishlist is a good idea.

**Comment by jaap-karssenberg** [comment]
In the latest version delete page always asks for confirmation, that should
elevate the issue a bit.

Still would be nice to have this feature to roll back page moves. But needs
design how to store all the relevant information.

Op zo 10 okt. 2021 18:09 schreef Nathaniel M. Beaver <
***@***.***>:
